gqlalchemy/loaders.py

```diff
@@ -0,0 +1,416 @@
+    variables: Optional[Dict[str, str]] = None
+
+
+
+@dataclass(frozen=True)
+class OneToManyMapping:
+    mapping: ForeignKeyMapping
+    from_entity: bool = False
+
+
+
+@dataclass(frozen=True)
+class ManyToManyMapping:
+    mapping1: ForeignKeyMapping
+    mapping2: ForeignKeyMapping
+    label: str
+    from_first: bool = True
+
+
+
+Mapping = Union[List[OneToManyMapping], ManyToManyMapping]
+
+
+
+@dataclass
+class TableMapping:
+    table_name: str
+    mapping: Optional[Mapping] = None
+    indices: Optional[List[str]] = None
+
+
+
+class DataSource(ABC):
+    def __init__(self, file_extension: str) -> None:
+        self._file_extension: str = file_extension
+
+    
+    @abstractmethod
+    def load_data(
+        self,
+        collection_name: str,
+        is_cross_table: bool=False
+    ) -> None:
+        pass
+
+
+
+class S3DataSource(DataSource):
+    def __init__(
+        self, 
+        bucket_name: str,
+        s3_access_key: str,
+        s3_secret_key: str,
+        s3_region: str,
+        file_extension: str,
+        s3_session_token: Optional[str] = None,
+    ) -> None:
+        super().__init__(file_extension=file_extension)
+        self._bucket_name: str = bucket_name
+        self._s3_access_key: str = s3_access_key
+        self._s3_secret_key: str = s3_secret_key
+        self._s3_region: str = s3_region
+        self._s3_session_token: Optional[str] = s3_session_token
+
+    
+    def load_data(
+        self,
+        collection_name: str,
+        is_cross_table: bool = False,
+        columns: Optional[List[str]] = None,
+    ) -> None:
+        s3 = fs.S3FileSystem(
+            region=self._s3_region,
+            access_key=self._s3_access_key,
+            secret_key=self._s3_secret_key,
+            session_token=self._s3_session_token,
+        )
+        source = f"{self._bucket_name}/{collection_name}.{self._file_extension}"
+        print("Loading " + ("cross table " if is_cross_table else "") + f"data from {source}")
+
+        # Load dataset via Pyarrow
+        dataset = ds.dataset(
+            source=source, 
+            filesystem=s3
+        )
+
+        # Load batches from raw data
+        for batch in dataset.to_batches(
+            columns=columns,
+        ):
+            for batch in batch.to_pylist():
+                yield batch
+
+
+@dataclass(frozen=True)
+class Configuration:
+    label: Optional[str] = None
+    column_names_mapping: Optional[Dict[str, str]] = None
+
+
+class TableToGraphTranslator:
+    _DIRECTION = {
+        True: ("a", "b"),
+        False: ("b", "a"),
+    }
+
+    def __init__(
+        self,
+        data_source: DataSource,
+        data_configuration: Dict[str, Any], 
+        memgraph: Optional[Memgraph] = None
+    ) -> None:
+        self._memgraph: Memgraph = memgraph if memgraph is not None else Memgraph()
+        self._data_source: DataSource = data_source
+
+        self.__load_configuration(data_configuration=data_configuration)
```

Comment:
Why are some methods prefixed with double underscore? One is by convention private, I'm not sure what are you trying to achieve? Is it some functionality that is done with __ when you type it in front?

Reply:
https://dbader.org/blog/meaning-of-underscores-in-python#:~:text=A%20double%20underscore%20prefix%20causes,the%20class%20is%20extended%20later.

Reply:
I want to keep it private, so when someone inherits this class it can't invoke double-underscore-prefixed methods

Reply:
ok sounds fine